src/main/java/com/verily/genomewarp/GenomeWarpSerial.java

```diff
@@ -798,35 +799,50 @@ private static VCFHeader warpHeader(VCFHeader in, Map<String, Long> namesAndLeng
       fail("failed to read from input bed: " + ex.getMessage());
     }
 
-    logger.log(Level.INFO, "Generating regions from variants");
-    Map<String, List<GenomeRange>> fromVcfBEDPerChromosome = generateBEDFromVCF(vcfReader);
+    Map<String, List<GenomeRange>> fromVcfBEDPerChromosome = null;
```

Comment:
Please factor these code paths into two separate functions for clarity. E.g.

Map<String, List<GenomeRange>> intermediateBedMap;
if (fragmentConfidentRegions) {
  intermediateBedMap = createFragmentedBed(vcfReader, dnaOnlyInputBEDPerChromosome);
} else {
  intermediateBedMap = dnaOnlyInputBedPerChromosome;
}

where the dnaOnlyInputBedPerChromosome has already had the sorting applied to it. Then you can loop through the chroms and do the massaging without an additional if clause, and it's clearer what exactly the new transformations do as it's all localized to a single function.